### PR TITLE
Make perturbation mode default

### PR DESCRIPTION
**Pull request by matt-frey**
Users have to provide the buoyancy perturbation instead of the total buoyancy.

### CODE REVIEW

File: src/fields/field_diagnostics_netcdf.f90 [hunk 1]
```diff
@@ -239,31 +234,34 @@ end subroutine write_netcdf_field_stats
...
 #endif
 
 #ifdef ENABLE_BUOYANCY
             call field_combine_physical(sbuoy, buoy)
             buf(8) = get_available_potential_energy(buoy, l_global=.false., l_allreduce=.false.)
 
-#ifdef ENABLE_PERTURBATION_MODE
+            ! get total buoyancy:
             do iz = 0, nz
                 buoy(iz, :, :) = buoy(iz, :, :) + bbarz(iz)
```
[the remark this answers is not in the document]
Comment (matt-frey) [reply src/fields/field_diagnostics_netcdf.f90:255]: Yes, it would. We can do that.

[the remark this answers is not in the document]
Comment (matt-frey) [reply src/fields/field_diagnostics_netcdf.f90]: Done.

File: src/fields/field_diagnostics_netcdf.f90 [hunk 2]
```diff
@@ -239,31 +234,34 @@ end subroutine write_netcdf_field_stats
...
             enddo
-#endif
+
             bmin = minval(buoy)
             bmax = maxval(buoy)
+
+            ! undo
+            do iz = 0, nz
+                buoy(iz, :, :) = buoy(iz, :, :) - bbarz(iz)
```
[the remark this answers is not in the document]
Comment (matt-frey) [reply src/fields/field_diagnostics_netcdf.f90]: Done.

File: src/fields/field_netcdf.f90 [hunk 3]
```diff
@@ -230,14 +226,22 @@ subroutine write_netcdf_fields(t)
...
 
-            do iz = 0, nz
-                buoy(iz, :, :) = buoy(iz, :, :) + bbarz(iz)
-            enddo
-#endif
-            call write_field_double(NC_BUOY, buoy, start, cnt)
+            if (nc_dset(NC_BUOY)%l_enabled) then
+                ! get total buoyancy
+                do iz = 0, nz
+                    buoy(iz, :, :) = buoy(iz, :, :) + bbarz(iz)
```
[the remark this answers is not in the document]
Comment (matt-frey) [reply src/fields/field_netcdf.f90]: Done.

File: src/fields/field_diagnostics_netcdf.f90 [hunk 4]
```diff
@@ -582,7 +574,6 @@ subroutine set_netcdf_stat_info
                 unit='m/s^2',                                           &
                 dtype=NF90_DOUBLE)
 
-#ifdef ENABLE_PERTURBATION_MODE
             nc_dset(NC_BASQ) = netcdf_stat_info(                        &
                 name='squared_buoyancy_anomaly',                        &
                 long_name='domain-averaged squared buoyancy anomaly',   &
```
[the remark this answers is not in the document]
Comment (matt-frey) [reply src/fields/field_diagnostics_netcdf.f90:577]: Yes, that is correct.

[the remark this answers is not in the document]
Comment (matt-frey) [reply src/fields/field_diagnostics_netcdf.f90]: Removed.

**Review by daviddritschel — APPROVED** [review]

**Pull request merged**